pdfminer/psparser.py

```diff
@@ -168,6 +169,16 @@ class PSBaseParser:
     def __init__(self, fp):
         self.fp = fp
         self.seek(0)
+        self.__spec_char_check_dict = {
+            b'%': (b'%', self._parse_comment),
+            b'/': (b'', self._parse_literal),
+            b'.': (b'.', self._parse_float),
+            b'<': (b'', self._parse_wopen),
+            b'>': (b'', self._parse_wclose),
+            b'(': (b'', self._parse_string),
+            b'-': (b'-', self._parse_number),
+            b'+': (b'+', self._parse_number)
+        }
```

Comment:
Smart, using a dict instead of if-else statement. Looking in a dict is constant which should be faster compared check every if-else statement. Maybe add some comment to make it more clear.

Reply:
Comments added by @amermsl92